### ISSUE TITLE
Compatibility with Oracle's S3 implementation

### ISSUE THREAD

**Issue by superfav** [trinodb/trino#14752]
Trino fails to create a table from a delta table stored in Oracle S3 with the following error message:
```
Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: The object 'test/_delta_log/00000000000000000001.json' was not found in the bucket 'xxxxx' (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey; Request ID: null; S3 Extended Request ID: null; Proxy: null)
```
`TransactionLogTail` expects the exception message to contain `The specified key does not exist`:
```java
        if (e.getMessage().contains("The specified key does not exist")) {
```
See [TransactionLogTail.java](../blob/d21cb3b/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TransactionLogTail.java#L158).
This matches the message returned by Amazon and other implementations:
```
com.amazonaws.services.s3.model.AmazonS3Exception: The specified key does not exist. (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey;
...
```
But it doesn't match the one returned by Oracle's implementation. Is it possible to change the test to check for the error code –`Error Code: NoSuchKey;`– instead?
```java
        if (e.getMessage().contains("Error Code: NoSuchKey;")) {
```
#### Stack trace:
```
io.trino.spi.TrinoException: Failed to access table location: s3a://xxxxx/test
	at io.trino.plugin.deltalake.metastore.HiveMetastoreBackedDeltaLakeMetastore.createTable(HiveMetastoreBackedDeltaLakeMetastore.java:163)
	at io.trino.plugin.deltalake.DeltaLakeMetadata.createTable(DeltaLakeMetadata.java:743)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.createTable(ClassLoaderSafeConnectorMetadata.java:374)
	at io.trino.metadata.MetadataManager.createTable(MetadataManager.java:654)
	at io.trino.execution.CreateTableTask.internalExecute(CreateTableTask.java:273)
	at io.trino.execution.CreateTableTask.execute(CreateTableTask.java:119)
	at io.trino.execution.CreateTableTask.execute(CreateTableTask.java:85)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:145)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:249)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_400____20221025_084620_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

...

Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: The object 'test/_delta_log/00000000000000000001.json' was not found in the bucket 'xxxxx' (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey; Request ID: null; S3 Extended Request ID: null; Proxy: null)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1862)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1415)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1384)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1154)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:811)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:779)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:753)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:713)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:695)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:539)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5453)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5400)
	at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1524)
	at io.trino.plugin.hive.s3.TrinoS3FileSystem$TrinoS3InputStream.lambda$openStream$2(TrinoS3FileSystem.java:1365)
	... 47 more
```

**Comment by ebyhr** [comment]
> Is it possible to change the test to check for the error code –Error Code: NoSuchKey;– instead?

The condition will be fixed in #12728

**Comment by superfav** [comment]
Thanks @ebyhr for taking care of this. I didn't know Ceph was also affected. Can I collaborate to close that PR? I see _codesorcery_ is in a similar situation about running a fork.

**Comment by krvikash** [comment]
Hi @superfav, Could you please check if https://github.com/trinodb/trino/commit/618b9e6c972c1ec5f15c68c1c6f9573aac719c16 would help in your case?

I see [comment](https://github.com/trinodb/trino/pull/12728#issuecomment-1495528710) from @codesorcery that everything works now with Ceph.

**Comment by superfav** [comment]
Hi @krvikash,
Thanks for the heads up. Looking at the code it doesn't look like it solves the issue. Like @codesorcery, I need to test it.
Thank you.

**Comment by superfav** [comment]
Hi @krvikash 

I can confirm it now works with Oracle Object Storage. Closing this issue as it's fixed by #16703.
